src/Sparrow/Json/BlittableJsonTextWriter.cs

```diff
@@ -173,13 +172,59 @@ public unsafe void WriteString(LazyStringValue str)
 
                 EnsureBuffer(1);
                 _buffer[_pos++] = Quote;
+
                 return;
             }
 
             UnlikelyWriteEscapeSequences(str, numberOfEscapeSequences, escapeSequencePos, strBuffer, size);
+
+            EnsureBuffer(1);
+            _buffer[_pos++] = Quote;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void WriteStringJoin(LazyStringValue[] strArray)
```

Comment:
In the metadata we support only strings values for now. 
We can either change this requirement or use string.Join and not use LazyStringValue for @attachments metadata.